### PR TITLE
Group Foxtrot Assignment 1

### PR DESCRIPTION
**Pull request by kz9**
Hello Prof Goodman,

This is our PR for assignment 1. Please inform us if there are any aspects that require modifications or improvements.

Best regards,
Team Foxtrot

### CODE REVIEW

File: services/backend/src/data_structures/user_based_recommender/foxtrot/UserBasedRecommender.py [hunk 1]
```diff
@@ -18,24 +22,88 @@ def __new__(cls):
 
     def gather_data(self):
         # Connect to the database and fetch user-content engagement.
+        if os.path.isfile("/usr/src/app/src/recommendation_system/ml_models/foxtrot/assets/colab.pkl"):
+            with open("/usr/src/app/src/recommendation_system/ml_models/foxtrot/assets/colab.pkl", "rb") as f:
+                self.interactions = pickle.load(f)
```
Comment (kennethgoodman) [inline services/backend/src/data_structures/user_based_recommender/foxtrot/UserBasedRecommender.py:27]: does this get overwritten every time?

File: services/backend/src/data_structures/user_based_recommender/foxtrot/UserBasedRecommender.py [hunk 2]
```diff
@@ -18,24 +22,88 @@ def __new__(cls):
...
         self.interactions = db.session.query(
             Engagement.content_id,
             Engagement.user_id,
             Engagement.engagement_type,
             Engagement.engagement_value
-        ).limit(5).all()
+        ).all()
+        with open("/usr/src/app/src/recommendation_system/ml_models/foxtrot/assets/colab.pkl", "wb") as file:
```
Comment (kennethgoodman) [inline services/backend/src/data_structures/user_based_recommender/foxtrot/UserBasedRecommender.py:34]: why are you dumping data here? remove this, it won't be used

**Review by kennethgoodman — APPROVED** [review]
please make changes and merge

File: services/backend/src/data_structures/user_based_recommender/foxtrot/UserBasedRecommender.py (hunk 2 above)
Comment (kz9) [reply services/backend/src/data_structures/user_based_recommender/foxtrot/UserBasedRecommender.py:34]: Sorry, I think the file is already in the assets folder. I will delete the rest of the code and just keep the loading file code.

**Pull request merged**